runtime/src/bank.rs

```diff
@@ -1791,6 +1792,21 @@ pub fn goto_end_of_slot(bank: &mut Bank) {
     }
 }
 
+// This protects from memory exhaustion.
+// Because we're are about to restore full process state from file, allow use
+// of memory as much as possible. But set a hard-limit with ample of head rooms.
+// This is an extra protection for possible bincode bugs in addition to
+// deserialize_snapshot_data_file
```

Comment:
I think this description could be made more concise.  Ideally explain it in one sentence.  A paragraph is too much to ask your reader for )
Also runtime/ should not reference things in ledger/  (the `deserialize_snapshot_data_file` reference) please -- like the code itself, avoid circular dependencies in code comments

Reply:
Thanks for pointing this! How about now?: 477695eb94a9a550e37ce05b15d65a3dc70a4c69